ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/CompletionDeduplicationInfoIT.scala

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.suites
+
+import com.daml.ledger.api.SubmissionIdGenerator
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.api.testtool.suites.CompletionDeduplicationInfoIT._
+import com.daml.ledger.api.v1.command_service.SubmitAndWaitRequest
+import com.daml.ledger.api.v1.command_submission_service.SubmitRequest
+import com.daml.ledger.api.v1.commands.Command
+import com.daml.ledger.api.v1.completion.Completion
+import com.daml.ledger.api.v1.ledger_configuration_service.LedgerConfiguration
+import com.daml.ledger.client.binding.Primitive
+import com.daml.ledger.test.model.Test.Dummy
+import com.daml.lf.data.Ref
+import com.daml.platform.testing.WithTimeout
+import com.google.protobuf.duration.Duration
+import io.grpc.Status
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+
+final class CompletionDeduplicationInfoIT(service: Service) extends LedgerTestSuite {
+
+  private val serviceName: String = service.productPrefix
+
+  override private[testtool] def name = serviceName + super.name
+
+  test(
+    shortIdentifier = s"CCDIIncludeDedupInfo$serviceName",
+    description = s"Deduplication information is preserved in completions ($serviceName)",
+    allocate(SingleParty),
+  )(implicit ec => { case Participants(Participant(ledger, party)) =>
+    for {
+      config <- ledger.configuration()
+      optApplicationIdCompletion <- submitRequest(service, ledger, party, simpleCreate(party))
+      optSubmissionIdCompletion <- submitRequest(
+        service,
+        ledger,
+        party,
+        simpleCreate(party),
+        updateCommandServiceRequest = _.update(_.commands.submissionId := aSubmissionId),
+        updateCommandSubmissionServiceRequest = _.update(_.commands.submissionId := aSubmissionId),
+      )
+      optCompletionDeduplicationTime <- submitRequest(
+        service,
+        ledger,
+        party,
+        simpleCreate(party),
+        updateCommandServiceRequest = _.update(_.commands.deduplicationTime := aDeduplicationTime),
+        updateCommandSubmissionServiceRequest =
+          _.update(_.commands.deduplicationTime := aDeduplicationTime),
+      )
+    } yield {
+      assertApplicationIdIsPreserved(ledger.applicationId, optApplicationIdCompletion)
+      assertSubmissionIdIsGenerated(optApplicationIdCompletion)
+      assertDefaultDeduplicationTimeIsReported(config, optApplicationIdCompletion)
+      assertSubmissionIdIsPreserved(aSubmissionId, optSubmissionIdCompletion)
+      assertDeduplicationTimeIsPreserved(aDeduplicationTime, optCompletionDeduplicationTime)
```

Comment:
Hm. This test won't work for Canton. Ledger implementations are free to convert the provided deduplication period into some other format.

Reply:
In particular, Canton converts everything into a deduplication offset.

Reply:
Moreover, ledger implementations are free to extend the deduplication period as they see fit. So even kvutils is allowed (and after some discusson with @miklos-da probably will) modify the dedup duration.

Reply:
Thanks for the feedback, then I think we should just remove it, unless the API offers more specific guarantees that can be tested (like which conversions are allowed, which I think it doesn't atm).

Reply:
Implementation-specific API tests about this may make sense, though.

Reply:
51e9b89